costs/costs_route.go

```diff
@@ -0,0 +1,205 @@
+//   Copyright 2017 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package costs
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/trackit/jsonlog"
+	"github.com/trackit/trackit2/db"
+	"github.com/trackit/trackit2/es"
+	"github.com/trackit/trackit2/routes"
+	"github.com/trackit/trackit2/users"
+)
+
+const (
+	iso8601TimeFormat = "2006-01-02"
+)
+
+// simpleCriterionMap will map simple criterion to the boolean true.
+// This will be used in parseCriterionQueryParams to validate the queryParam.
+// It does not take into account the 'tag:*' criterion as it is not fixed.
+var simpleCriterionMap = map[string]bool{
+	"year":    true,
+	"month":   true,
+	"week":    true,
+	"day":     true,
+	"account": true,
+	"product": true,
+	"region":  true,
+}
+
+// esQueryParams will store the parsed query params
+type esQueryParams struct {
+	dateBegin         time.Time
+	dateEnd           time.Time
+	accountList       []string
+	aggregationParams []string
+}
+
+type queryParamsParser func([]string, esQueryParams) (esQueryParams, error)
+
+// formFieldsToHandleFunc maps a query param to a queryParamsParser function
+// to parse it.
+var formFieldsToHandlerFunc = map[string]queryParamsParser{
+	"since":    parseSinceQueryParam,
+	"until":    parseUntilQueryParam,
```

Comment:
Usually we refer to these as `begin` and `end`. Please change to use those terms.